src/components/AccountLists/AccountLists.tsx

```diff
@@ -85,173 +85,154 @@ const AccountLists = ({ data }: Props): ReactElement => {
       >
         <Container>
           <Grid container spacing={3}>
-            {data.accountLists.nodes.map(
-              ({
+            {data.accountLists.nodes.map((accountList) => {
+              const {
                 id,
                 name,
-                monthlyGoal: preferencesGoal,
-                monthlyGoalUpdatedAt: preferencesGoalUpdatedAt,
                 receivedPledges,
                 totalPledges,
-                currency: preferencesCurrency,
+                currency,
                 healthIndicatorData,
-              }) => {
-                const hasPreferencesGoal = typeof preferencesGoal === 'number';
-                const monthlyGoal = hasPreferencesGoal
-                  ? preferencesGoal
-                  : healthIndicatorData?.machineCalculatedGoal;
-                const currency = hasPreferencesGoal
-                  ? preferencesCurrency
-                  : healthIndicatorData?.machineCalculatedGoalCurrency;
-                const hasMachineCalculatedGoal =
-                  !hasPreferencesGoal && typeof monthlyGoal === 'number';
-                const preferencesGoalDate =
-                  typeof preferencesGoal === 'number' &&
-                  preferencesGoalUpdatedAt &&
-                  DateTime.fromISO(preferencesGoalUpdatedAt);
+              } = accountList;
 
-                // If the currency comes from the machine calculated goal and is different from the
-                // user's currency preference, we can't calculate the received or total percentages
-                // because the numbers are in different currencies
-                const hasValidGoal =
-                  currency === preferencesCurrency && !!monthlyGoal;
-                const receivedPercentage = hasValidGoal
-                  ? receivedPledges / monthlyGoal
-                  : NaN;
-                const totalPercentage = hasValidGoal
-                  ? totalPledges / monthlyGoal
-                  : NaN;
+              const {
+                goal,
+                goalSource,
+                preferencesGoalUpdatedAt,
+                preferencesGoalLow,
+                preferencesGoalOld,
+              } = getHealthIndicatorInfo(accountList, healthIndicatorData);
 
-                const annotation: Annotation | null = hasMachineCalculatedGoal
-                  ? {
-                      label: t('machine-calculated'),
-                      color: 'statusWarning.main',
-                    }
-                  : preferencesGoalDate
-                  ? {
-                      label: t('Last updated {{date}}', {
-                        date: dateFormat(preferencesGoalDate, locale),
-                      }),
-                      variant: 'body2',
-                    }
-                  : null;
-                const annotationId = `annotation-${id}`;
+              const hasValidGoal = goal !== null;
```

Comment:
Do we need to update the healthIndicator to remove the export mismatchedCurrencies?
I guess it would be nice to keep it for testing

Reply:
Good point that it's not used anymore